tests/kernel/usb_drive.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: usb_nic
+# Summary: Simple smoke test for testing USB drive connected to system
+# Maintainer: LSG QE Kernel <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my ($self) = @_;
+
+    select_serial_terminal;
+
+    my $lun = script_output 'lsscsi -t -v | awk -F" " \'/usb/ {split($2,a,/[\/]/); print a[6]}\'';
+    die "no usb storage device connected" if $lun eq "";
+
+    my $device = "/dev/" . script_output "lsscsi -v | awk -F\"/\" \'/$lun/ {print \$3; exit}\'";
+
+    # create filesystem, mountpoint and temporary file
+    my $tmp = script_output 'mktemp -d';
+    my $file = "$tmp/file";
+    my $md5 = "$tmp/md5";
+    my $mountpoint = "$tmp/mount";
+    my $file_copy = "$mountpoint/file";
+
+    assert_script_run "mkdir $mountpoint";
+    assert_script_run "mkfs.btrfs -f $device";
+
+    assert_script_run "mount -t btrfs $device $mountpoint";
+    assert_script_run "dd if=/dev/urandom of=$file bs=1M count=16";
+    assert_script_run "md5sum $file > $md5";
+    assert_script_run "cp $file $file_copy";
+
+    # sync, unmount and flush slab and page cache
+    assert_script_run 'sync';
+    assert_script_run "umount $mountpoint";
```

Comment:
It might be better to drop the `sync` because it should be done automatically by `umount`.